drivers/gpio/gpio_esp32.c

```diff
@@ -148,17 +148,27 @@ static int gpio_esp32_config(struct device *dev, int access_op,
 
 	pinmux_pin_set(data->pinmux, pin, PIN_FUNC_GPIO);
 	if (flags & GPIO_PUD_PULL_UP) {
-		pinmux_pin_pullup(data->pinmux, pin, PINMUX_PULLUP_ENABLE);
+		r = pinmux_pin_pullup(data->pinmux, pin, PINMUX_PULLUP_ENABLE);
```

Comment:
For this operation, there's no need to check: it's going to fail only if the particular pin number is invalid as far as GPIO pins in ESP32 are concerned. (`pinmux_pin_get()` above checks this.)